### DISCUSSION TITLE
Trigger data-sources sync via API

### DISCUSSION THREAD

**Discussion by jhofmueller** [netbox-community/netbox#13926] [Help Wanted!] [answered]
Hi!

I have a git repository as data-source for custom scripts. Every time I change something in the script I have to manuall tigger syncing. I would like to automate that so that when a change get's commited and the tests succeed git (we use gitlab here) uses a webhook to trigger a sync for the data-source.

Grateful for any ideas and pointers.

j.

**Comment by LuPo** [comment, marked as answer]
You could create an external webhook receiver that, when triggered, instructs netbox through API . I did not wanted to maintain an external webhook receiver and define an action to be executed. For this reason I have created a Netbox Plugin Webhook Receiver. 

https://netbox.dev/plugins/netbox-plugin-webhook-receiver/

Another way you could deal with this issue is to execute `netbox/manage.py syncdatasource` on server scheduling it with crontab.
This issue can be of some help: #12464
